program-error/src/lib.rs

```diff
@@ -122,12 +124,14 @@ impl fmt::Display for ProgramError {
     }
 }
 
+#[cfg(feature = "num-traits")]
 pub trait PrintProgramError {
     fn print<E>(&self)
     where
         E: 'static + std::error::Error + DecodeError<E> + PrintProgramError + FromPrimitive;
```

Comment:
Is there a way to relax the requirements for `E`? E.g., drop `std::error::Error` and `FromPrimitive`.